docs/proposals/koordlet/20231208-support-netqos.md

```diff
@@ -0,0 +1,395 @@
+When the load is low, the network bandwidth for low-priority(`l2`) containers is accommodated as much as possible.
+
+we will consider expanding the fields of `nodeslo` to add new parameters related to network bandwidth as follows:
+```go
+type NodeSLOSpec struct {
+	// QoS config strategy for pods of different qos-class
+	ResourceQOSStrategy *ResourceQOSStrategy `json:"resourceQOSStrategy,omitempty"`
+	//node global system config
+	SystemStrategy *SystemStrategy `json:"systemStrategy,omitempty"`
+}
+
+type ResourceQOSStrategy struct {
+	// Policies of pod QoS.
+	Policies *ResourceQOSPolicies `json:"policies,omitempty"`
+
+	// ResourceQOS for LSR pods.
+	LSRClass *ResourceQOS `json:"lsrClass,omitempty"`
+
+	// ResourceQOS for LS pods.
+	LSClass *ResourceQOS `json:"lsClass,omitempty"`
+
+	// ResourceQOS for BE pods.
+	BEClass *ResourceQOS `json:"beClass,omitempty"`
+
+	// ResourceQOS for system pods
+	SystemClass *ResourceQOS `json:"systemClass,omitempty"`
+
+	// ResourceQOS for root cgroup.
+	CgroupRoot *ResourceQOS `json:"cgroupRoot,omitempty"`
+}
+
+type ResourceQOS struct {
+	...
+	NetworkQOS *NetworkQOSCfg `json:"networkQOS,omitempty"`
+}
+
+type NetworkQOSCfg struct {
+	Enable     *bool `json:"enable,omitempty"`
+	NetworkQOS `json:",inline"`
+}
+
+type NetworkQOS struct {
+	// IngressRequest describes the minimum network bandwidth guaranteed in the ingress direction.
+	// unit: bps(bytes per second), two expressions are supported，int and string,
+	// int: percentage based on total bandwidth，valid in 0-100
+	// string: a specific network bandwidth value, eg: 50M.
+	// +kubebuilder:default=0
+	IngressRequest *intstr.IntOrString `json:"ingressRequest,omitempty"`
+	// IngressLimit describes the maximum network bandwidth can be used in the ingress direction,
+	// unit: bps(bytes per second), two expressions are supported，int and string,
+	// int: percentage based on total bandwidth，valid in 0-100
+	// string: a specific network bandwidth value, eg: 50M.
+	// +kubebuilder:default=100
+	IngressLimit *intstr.IntOrString `json:"ingressLimit,omitempty"`
+
+	// EgressRequest describes the minimum network bandwidth guaranteed in the egress direction.
+	// unit: bps(bytes per second), two expressions are supported，int and string,
+	// int: percentage based on total bandwidth，valid in 0-100
+	// string: a specific network bandwidth value, eg: 50M.
+	// +kubebuilder:default=0
+	EgressRequest *intstr.IntOrString `json:"egressRequest,omitempty"`
+	// EgressLimit describes the maximum network bandwidth can be used in the egress direction,
+	// unit: bps(bytes per second), two expressions are supported，int and string,
+	// int: percentage based on total bandwidth，valid in 0-100
+	// string: a specific network bandwidth value, eg: 50M.
+	// +kubebuilder:default=100
+	EgressLimit *intstr.IntOrString `json:"egressLimit,omitempty"`
+}
+
+type SystemStrategy struct {
```

Comment:
Is this a cluster level configuration?

Reply:
yep

Reply:
> yep

If different Nodes in a cluster have different bandwidth performance, how to deal with this?

Reply:
here are some defination: 
https://github.com/koordinator-sh/koordinator/blob/8a2d393/apis/configuration/slo_controller_config.go#L87-L91